lib/python/picongpu/picmi/gaussian_laser.py

```diff
@@ -42,31 +84,56 @@ def get_as_pypicongpu(self) -> laser.GaussianLaser:
         # unsupported: fill_in (do not warn, b/c we don't know if it has been
         # set explicitly, and always warning is bad)
 
-        assert self.centroid_position[0] == self.focal_position[0] \
-            and self.centroid_position[2] == self.focal_position[2], \
-            "focal position x and z MUST be equal to centroid x and z"
-        assert [0, 1, 0] == self.propagation_direction, \
-            "only support propagation along Y axis"
-        assert 0 == self.centroid_position[1], "centroid MUST have y=0"
+        assert abs(self.scalarProduct(self.polarization_direction,
+                self.polarization_direction) - 1) < 1e-9, \
+            "the polarization direction vector must be normalized"
+
+        # check for excessive phase values to avoid numerical precision errors
+        assert abs(self.picongpu_phase) <= 2*3.1416, \
+            "abs(phase) must be < 2*pi"
+
+        # check that initialising from y_min-plane only is sensible
+        assert (self.scalarProduct(self.propagation_direction, \
+                [0., 1., 0.]) > 0.), \
+            "laser propagation parallel to the y-plane or pointing outside \
+            from the inside of the simulation box is not supported by this \
+            laser in PIConGPU"
+        assert ((self.propagation_direction[0]**2
+                + self.propagation_direction[1]**2
+                + self.propagation_direction[2]**2) - 1. < 1e-9), \
```

Comment:
Is there something like `epsilon` in python available that we do not need to use a magic number `1e-9` here?

Reply:
Yes, but it requires another non negligibel dependency
```
>>> import sys
>>> sys.float_info.epsilon
2.220446049250313e-16
```
Also I am not sure whether we want to test against machine epsilon necessarily.

Reply:
I could parametrize it if you prefer?

Reply:
done